### PR TITLE
Add FOR SHARE LOCK to FOR UPDATE OF clause

### PR DESCRIPTION
**Pull request by yeongseon**

### CODE REVIEW

File: test/test_suite.py [hunk 1]
```diff
@@ -320,3 +322,33 @@ def test_pass_uri_query_as_kwargs(self):
...
         assert kwargs["encrypt"] == "true"
         assert kwargs["compress"] == "true"
+
+
+class  HANACompileTest(fixtures.TestBase, AssertsCompiledSQL):
+
+    __dialect__ = testing.db.dialect
+
+    @testing.only_on('hana')
+    @testing.only_if('hana+hdbcli')
```
Comment (jarus) [inline test/test_suite.py]:
```suggestion
```

As the test is SQL specific, it should also work other HANA client libraries.

File: test/test_suite.py [hunk 2]
```diff
@@ -320,3 +322,33 @@ def test_pass_uri_query_as_kwargs(self):
...
+    def test_for_update(self):
```
Comment (jarus) [inline test/test_suite.py]:
```suggestion
    def test_sql_row_locking(self):
```

File: sqlalchemy_hana/dialect.py [hunk 3]
```diff
@@ -69,16 +69,19 @@ def limit_clause(self, select, **kwargs):
         return text
 
     def for_update_clause(self, select, **kwargs):
-        tmp = " FOR UPDATE"
+        if select._for_update_arg.read:
+            tmp = " FOR SHARE LOCK"
```
Comment (jarus) [inline sqlalchemy_hana/dialect.py]: In the current implementation, the dialect would silently ignore the other parameters like `NOWAIT` or ` OF ...`. What do you think, should the dialect prevent wrong usage and raise an exception that is visible the applications test suite or while development?

Comment (jarus) [reply sqlalchemy_hana/dialect.py]: Valid point. If the other dialects silently ignore the error then we should follow. Maybe add a small comment about that into the source code. :)

File: test/test_suite.py (hunk 1 above)
Comment (jarus) [reply test/test_suite.py]: It should highlight that this test is only valid/useful for the HANA dialect but it is probably not required.

File: sqlalchemy_hana/dialect.py (hunk 3 above)
Comment (yeongseon) [reply sqlalchemy_hana/dialect.py]: I added some comments here. what do you think?

Comment (jarus) [reply sqlalchemy_hana/dialect.py]: Looks good to me.

**Pull request merged**